File: app_logger.py
import os


class AppLogger:
    def __init__(self, message, file_name="app.log", empty_line=False):
        if not message:
            raise ValueError("Invalid message")

        self.message = message
        self.file_path = os.path.join("log", file_name)
        self.empty_line = empty_line

    @classmethod
    def call(cls, message, file_name="app.log", empty_line=False):
        logger = cls(message, file_name, empty_line)
        logger.log()

    def log(self):
        os.makedirs(os.path.dirname(self.file_path), exist_ok=True)
        with open(self.file_path, "a") as file:
            self._print_and_write_to_log(file)

    def _print_and_write_to_log(self, file):
        if self.empty_line:
            print(self.message)
            print()

            file.write(self.message + "\n")
            file.write("\n")
        else:
            print(self.message)

            file.write(self.message + "\n")

File: app_logger.rb
# frozen_string_literal: true

require 'pry-byebug'

class AppLogger
  def self.call(message, options = {})
    new(message, options).call
  end

  attr_reader :message, :file_path, :empty_line

  def initialize(message, options)
    raise ArgumentError, 'Invalid message' if message.empty?

    @message = message

    file_name = options[:file_name] || 'app.log'
    @file_path = File.join('log', file_name)

    @empty_line = options[:empty_line]
  end

  def call
    File.open(file_path, 'a') { |file| puts_and_write_to_log(file) }
  end

  private

  def puts_and_write_to_log(file)
    puts(message)
    file.puts(message)
    return unless empty_line

    puts
    file.puts
  end
end
